pkg/engine2/reconciler/remove_edge.go

```diff
@@ -47,7 +47,10 @@ func RemovePath(
 	// We will pass the explicit field as false so that explicitly added resources do not get deleted
 	for _, path := range paths {
 		for _, resource := range path {
-			errs = errors.Join(errs, RemoveResource(ctx, resource, false))
+			err := RemoveResource(ctx, resource, false)
+			if err != nil && !errors.Is(err, graph.ErrVertexNotFound) {
```

Comment:
When could this return `ErrVertexNotFound`? That'd be a surprising result without knowing more. Can you add a comment how that could happen and why it's okay to ignore?